assignments/arrays.js

```diff
@@ -63,32 +63,80 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+arr = inventory;
+
+for (let i = 0; i < arr.length; i++) {
+
+    if(arr[i].id === 33) {
+
+        console.log(`Car 33 is a ${arr[i].car_make} ${arr[i].car_model} ${arr[i].car_year}` );
+    }
+}
+
+
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
 let lastCar = 0;
-console.log();
+ 
+lastCar = arr[arr.length - 1];
+    
+console.log(lastCar);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+
+inventory.sort(function(a, b){
+    var x = a.car_model.toLowerCase();
+    var y = b.car_model.toLowerCase();
+    if (x < y) {return -1;}
+    if (x > y) {return 1;}
+    return 0;
+
+  });
+
+console.log(inventory);
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+
+carY = inventory;
```

Comment:
I notice you continually reassign `carY` to `inventory`. Here's the thing: `carY` is in the global scope, so once you assign `carY` to be the same array as `inventory`, it will _always_ be the same array as `inventory` until either you change it or we reach the end of the file. 

If you were doing this to reset `carY` back to what `inventory` was, that isn't what's happening. As we discussed before, `carY` and `inventory` are pointing to the _same_ array in memory, so changing one changes the other.

Also, depending on what else is happening on the file, using implicit variables (_i.e. not using `var`, `let`, or `const`_) can lead to hard-to-find bugs. Using a variable identifier is a good idea, and I strongly suggest you use only `const` or `let`.

Reply:
Ok I see. I'll try to find better ways to rename my arrays.